### PR TITLE
86 header item component

### PR DESCRIPTION
**Pull request by mindfulme**
#### What's this PR do?
Adds header item component for the sortable table
#### Where should the reviewer start?
at HeaderItem.vue
#### How should this be manually tested?
The component is placed at components page
#### Any background context you want to provide?
There is a note from Armen that the sorting arrows should be placed leftward the title
photo_2018-10-19_11-15-30

#### What are the relevant tickets?
[#86](https://github.com/bitshares/bitshares-community-ui/issues/86)
#### Screenshots (if appropriate)
screen shot 2018-10-21 at 09 59 08

#### URL
Provide a link to your deploy from http://staging-ui.trusty.apasia.tech:8080/
[http://staging-ui.trusty.apasia.tech:8080/86-header-item-component ](http://staging-ui.trusty.apasia.tech:8080/86-header-item-component)
#### Questions:
What can be improved?

### CODE REVIEW

File: src/components/HeaderItem/HeaderItem.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
+Examples:
+ 
+Header item sorted in ascending way
+```js
+        <HeaderItem sort="asc">
+          <div slot="title">
```
Comment (roma219) [inline src/components/HeaderItem/HeaderItem.md]: Why use slot for this? Title is a simple string, `prop` is enough

File: src/components/HeaderItem/HeaderItem.vue [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+  color: config('colors.sort-arrow');
+}
+
+.sort-arrow-up {
+  @apply mb-2;
+  height: 3px;
+}
+
+.sort-arrow-up:not(.active):hover {
+  color: config('colors.sort-arrow-hover');
```
Comment (roma219) [inline src/components/HeaderItem/HeaderItem.vue]: is this really some new unique color not used anywhere else? as well as other arrows colors?

File: src/components/HeaderItem/HeaderItem.vue [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+<style lang="scss" scoped>
+.header-item {
+  font-size: config('textSizes.xs-sm');
+  display: flex;
+  color: config('colors.sort-arrow');
+}
+
+.sort-arrow-up {
+  @apply mb-2;
+  height: 3px;
```
Comment (roma219) [inline src/components/HeaderItem/HeaderItem.vue]: no px - use rem. https://github.com/MMF-FE/vue-svgicon - how to set width/height of the icon

File: src/components/HeaderItem/HeaderItem.vue [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+  </div>
+</template>
+
+<script>
+import '@icons/sortArrow'
+
+export default {
+  props: {
+    sort: {
+      default: () => null,
```
Comment (roma219) [inline src/components/HeaderItem/HeaderItem.vue]: why it defauts to function returning null? Why not just empty string?

**Review by roma219 — CHANGES_REQUESTED** [review]
Needs changes

Also:
1) Rename component to SortableHeaderItem
2) Title should change colors depending on if there is any sort applying or not

File: src/components/HeaderItem/HeaderItem.vue (hunk 2 above)
Comment (mindfulme) [reply src/components/HeaderItem/HeaderItem.vue]: Should be used somewhere else, I previously assumed that it should have the color which describes the element where the color is used

File: src/components/HeaderItem/HeaderItem.vue (hunk 3 above)
Comment (mindfulme) [reply src/components/HeaderItem/HeaderItem.vue]: this is useful

File: src/components/HeaderItem/HeaderItem.vue (hunk 4 above)
Comment (mindfulme) [reply src/components/HeaderItem/HeaderItem.vue]: In the issue, it is said to return `null` by default, probably I should have asked about it. Hmm, I guess I got it

File: src/components/HeaderItem/HeaderItem.md (hunk 1 above)
Comment (mindfulme) [reply src/components/HeaderItem/HeaderItem.md]: Okay, though some other content could be passed (using html tags)

File: src/components/HeaderItem/HeaderItem.vue (hunk 3 above)
Comment (mindfulme) [reply src/components/HeaderItem/HeaderItem.vue]: though they use pixels too
screen shot 2018-10-23 at 23 39 49

Comment (mindfulme) [reply src/components/HeaderItem/HeaderItem.vue]: or rather pass from props
screen shot 2018-10-23 at 23 42 43

File: src/components/SortableHeaderItem/SortableHeaderItem.vue [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+  </div>
+</template>
+
+<script>
+import '@icons/sortArrow'
+
+export default {
+  props: {
+    sort: {
+      default: () => '',
```
Comment (roma219) [inline src/components/SortableHeaderItem/SortableHeaderItem.vue]: just `default: ''` is enough
https://vuejs.org/v2/guide/components.html

**Review by roma219 — CHANGES_REQUESTED** [review]
Small cleanup and ok

**Review by roma219 — APPROVED** [review]

**Pull request merged**